src/Type/Generic/GenericClassStringType.php

```diff
@@ -172,4 +173,24 @@ public static function __set_state(array $properties): Type
 		return new self($properties['type']);
 	}
 
+	public function tryRemove(Type $typeToRemove): ?Type
+	{
+		if ($typeToRemove instanceof ConstantStringType && $typeToRemove->isClassString()) {
+			$generic = $this->getGenericType();
+
+			if ($generic instanceof ObjectType) {
```

Comment:
TypeWithClassName pls

Reply:
Intried that while PR creation and it made some tests fail.

Therefore I went back to ObjectType

Reply:
That's a red flag, we should investigate that

Reply:
ok - seems to work now. maybe it was some intermediate step which made it fail…